src/instruction.rs

```diff
@@ -1208,11 +1208,17 @@ impl Instruction {
                     FrameMatchCondition::AnyOfNames(frame_names),
                 ])
             }),
-            Instruction::Fence(Fence { qubits }) => Some(if qubits.is_empty() {
-                FrameMatchCondition::All
-            } else {
-                FrameMatchCondition::AnyOfQubits(Cow::Borrowed(qubits))
-            }),
+            Instruction::Fence(Fence { qubits }) => {
```

Comment:
Is this change for a separate bug?

Reply:
It's related, actually - this change is necessary to get the right behavior from the test cases. Prior to this, fences "used" frames which meant that they were serialized against one another, which can throw the timing of a program off because it can serialize two otherwise concurrent instructions. 

This change fixes it so FENCE only blocks frames, which makes more intuitive sense as well.